src/blenderbim/blenderbim/tool/qto.py

```diff
@@ -35,3 +40,113 @@ def get_radius_of_selected_vertices(cls, obj):
     @classmethod
     def set_qto_result(cls, result):
         bpy.context.scene.BIMQtoProperties.qto_result = str(round(result, 3))
+
+    @classmethod
+    def set_active_object(cls, object):
+        bpy.context.view_layer.objects.active = object
+
+    @classmethod
+    def get_pset_qto_object_ifc_info(cls, object):
+        element = tool.Ifc.get_entity(object)
+        pset_qto_ifc_instance = ifcopenshell.util.element.get_psets(element, qtos_only = True)
+        return pset_qto_ifc_instance
+
+    @classmethod
+    def get_pset_qto_properties(cls, object):
+        file = tool.Ifc.get()
+        schema = file.schema
+        pset_qto = util.pset.PsetQto(schema)
+        pset_qto_name = cls.get_pset_qto_name(object)
+        pset_qto_properties = pset_qto.get_by_name(pset_qto_name).get_info()['HasPropertyTemplates']
+        return pset_qto_properties
+
+    @classmethod
+    def get_pset_qto_name(cls, object):
+        applicable_pset_names = cls.get_applicable_pset_names(object)
+        for applicable_pset_name in applicable_pset_names:
+            if 'Qto_' in applicable_pset_name:
+                pset_qto_name = applicable_pset_name
+                return pset_qto_name
+
+    @classmethod
+    def get_applicable_pset_names(cls, object):
+        file = tool.Ifc.get()
+        schema = file.schema
+        pset_qto = util.pset.PsetQto(schema)
+        entity = tool.Ifc.get_entity(object)
+        ifc_object_type = entity.get_info()['type']
+        applicable_pset_names = pset_qto.get_applicable_names(ifc_object_type)
+        return applicable_pset_names
+
+    @classmethod
+    def edit_qto(cls, object, calculated_quantities):
+        file = tool.Ifc.get()
+        pset_qto_id = cls.get_pset_qto_id(object)
+        pset_qto_name = cls.get_pset_qto_name(object)
```

Comment:
This looks overly complex. What's going on here?

Reply:
Maybe is it better if the name of this function was "_assign_calculated_quantities_to_object_"?
Because it actually doesn't only edit the qto but it also get the pset_qto assigned to object in order to assign quantities to correct pset..
Am i missing something? :-)